src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php

```diff
@@ -107,37 +122,53 @@ public function generateClassStub(string $className) : ?StubData
 
     public function generateFunctionStub(string $functionName) : ?StubData
     {
-        if (! array_key_exists($functionName, PhpStormStubsMap::FUNCTIONS)) {
+        $lowercaseFunctionName = strtolower($functionName);
+
+        if (! array_key_exists($lowercaseFunctionName, self::$functionMap)) {
             return null;
         }
 
-        $filePath = PhpStormStubsMap::FUNCTIONS[$functionName];
+        $filePath = self::$functionMap[$lowercaseFunctionName];
 
-        if (! array_key_exists($functionName, $this->functionNodes)) {
+        if (! array_key_exists($lowercaseFunctionName, $this->functionNodes)) {
             $this->parseFile($filePath);
         }
 
-        return new StubData($this->createStub($this->functionNodes[$functionName]), $this->getExtensionFromFilePath($filePath));
+        return new StubData($this->createStub($this->functionNodes[$lowercaseFunctionName]), $this->getExtensionFromFilePath($filePath));
     }
 
     public function generateConstantStub(string $constantName) : ?StubData
     {
-        // https://github.com/JetBrains/phpstorm-stubs/pull/591
-        if (in_array($constantName, ['TRUE', 'FALSE', 'NULL'], true)) {
-            $constantName = strtolower($constantName);
+        $lowercaseConstantName = strtolower($constantName);
+
+        if (! array_key_exists($lowercaseConstantName, self::$constantMap)) {
+            return null;
         }
 
-        if (! array_key_exists($constantName, PhpStormStubsMap::CONSTANTS)) {
+        if (array_key_exists($lowercaseConstantName, $this->constantNodes)
+            && $this->constantNodes[$lowercaseConstantName] === null
```

Comment:
why cam this be `null`? I see the type was expanded for that...

Reply:
Constants are not case insensitive by default. They are case insensitive only if you use third parameter of `define`.

So this is optimization (line 165) if do more searches for the same constant that is not case insensitive.

Reply:
That still doesn't clarify the `null` scenario 🤔

Reply:
Eg. `DATE_ATOM` is not case insensitive.

First search for `date_atom` will call `->parseFile()`. It will save `date_atom` as `null` so second search for `date_atom` will not parse file again.